src/bin/edit/draw_filepicker.rs

```diff
@@ -113,10 +113,10 @@ pub fn draw_file_picker(ctx: &mut Context, state: &mut State) {
 
             // Check if the file already exists and show an overwrite warning in that case.
             if state.wants_file_picker != StateFilePicker::Open
-                && let Some(path) = doit.as_deref()
+                && doit.is_some()
                 && let Some(doc) = state.documents.active()
                 && let Some(file_id) = &doc.file_id
-                && sys::file_id_at(path).is_ok_and(|id| &id == file_id)
+                && sys::file_id(None, doit.as_deref()).is_ok_and(|id| &id == file_id)
```

Comment:
If `doit` is `None`, `file_id(None,None)` will fail with an error. Technically this means we can remove line R116. WDYT?